diff --git a/wscript b/wscript
index 22b732f..e8e3b8c 100644
--- a/wscript
+++ b/wscript
@@ -121,7 +121,7 @@ def configure(ctx):
         os.environ["PKG_CONFIG_PATH"] = 'packaging/win32_3rdparty/lib/pkgconfig'
         os.environ["PKG_CONFIG_LIBDIR"] = os.environ["PKG_CONFIG_PATH"]
 
-    elif ctx.options.WITH_STATIC_EXAMPLES and sys.platform.startswith('linux' or sys.platform == 'darwin'):
+    elif ctx.options.WITH_STATIC_EXAMPLES and sys.platform.startswith('linux') or sys.platform == 'darwin':
         print ("→ Compiling with static examples on Linux: search for pre-built dependencies in 'packaging/debian'")
         os.environ["PKG_CONFIG_PATH"] = 'packaging/debian_3rdparty/lib/pkgconfig'
         os.environ["PKG_CONFIG_LIBDIR"] = os.environ["PKG_CONFIG_PATH"]
